test/library/packages/UnitTest/TestResult.chpl

```diff
@@ -0,0 +1,123 @@
+/*
+  Holder for test result information.
+  Test results are automatically managed by the TestRunner, and do not 
+  need to be explicitly manipulated by writers of tests.  
+  Each instance holds the total number of tests run, and collections of 
+  failures and errors that occurred among those test runs. The collections
+  contain tuples of (testcase, exceptioninfo), where exceptioninfo is the
+  formatted traceback of the error that occurred.
+*/
+class TestResult {
+  type tupType = 2*string;
+  var failures: [1..0] tupType,
+      errors: [1..0] tupType,
+      skipped: [1..0] tupType;
+  var testsRun = 0;
+  var shouldStop = false;
+  var separator1 = "="* 70,
+      separator2 = "-"* 70;
+  // called when a test if about to run
+  proc startTest() {
+    this.testsRun += 1;
+  }
+  
+  /*Called when an error has occurred.*/
+  proc addError(test, errMsg) {
+    this.errors.push_back((test: string, errMsg: string));
+  }
+
+  /*called when error occured */
+  proc addFailure(test, errMsg) {
+    this.failures.push_back((test: string, errMsg: string));
+  }
+
+  /*Called when a test has completed successfully*/
+  proc addSuccess(test) { }
+
+  /*Called when a test is skipped.*/
+  proc addSkip(test, reason) {
+    this.skipped.push_back((test: string, reason: string));
+  }
+
+  /*Tells whether or not this result was a success.*/
+  proc wasSuccessful() {
+    return this.failures.size == 0 && this.errors.size == 0;
+  }
+  
+  /* Indicates that the tests should be aborted. */
+  proc stop() {
+    this.shouldStop = true;
+  }
+
+  /*Count of test skipped*/
+  proc skippedTests() {
```

Comment:
Design consideration: `skippedTests` might imply that it returns the array of skipped tests themselves. `numSkippedTests` would be less ambiguous. However, I don't feel too strongly about it.

Reply:
I think you are correct. I have made necessary changes.